File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-6.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int n = nums.size(), l = 0, r = n - 1;
        while (l<=r){
            int mid = l+(r-l)/2;
            if(target==nums[mid]){
                return mid;
            }
            // Left half is sorted
            if (nums[l] <= nums[mid]) {
                if (nums[l] <= target && target < nums[mid])
                    r = mid - 1;
                else
                    l = mid + 1;
            }else{
                 if (nums[r] >= target && target > nums[mid])
                    l = mid + 1;
                else
                    r = mid - 1;
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/jump-game-ii/submission-6.py
class Solution:
    def jump(self, nums: List[int]) -> int:
 
        n=len(nums)
        dp=[n]*(n+1)
        dp[0]=0
        for i in range(n):
            for j in range(i,i+nums[i]+1):
                if j < n:
                    dp[j]=min(dp[j],dp[i]+1)
        return dp[n-1]